app/frontend/src/lib/events.js

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+import logger from './logging';
+import { railsCsrfToken } from './utils';
+
+const EVENTS_ENDPOINT = '/api/events';
+
+// Triggers an event and makes a request to the backend endpoint with the event data.
+export const triggerEvent = (type, data) => {
+  const headers = { 'X-CSRF-Token': railsCsrfToken() };
+
+  return new Promise((resolve) => {
+    axios.post(EVENTS_ENDPOINT, { type, data }, { headers })
```

Comment:
i think this should return the axios promise to the outer one?

Reply:
The idea is that this promise always resolves, regardless of whether or not the Axis promise does. If Axios fails we still want to resolve not reject (but log the error to Rollbar/Sentry/you name it).